components/event-bus/cmd/event-bus-publish/handlers/handlers.go

```diff
@@ -90,6 +90,11 @@ func handlePublishRequest(w http.ResponseWriter, r *http.Request, publisher *con
 		return
 	}
 	publishRequest, err := parseRequest(body)
+
+	if len(publishRequest.SourceID) == 0 {
```

Comment:
At this point `err` may not be nil.

Please move this check after calling `ValidatePublish`.

Reply:
Thanks for the feedback. As discussed the logic will be added after `err` nil check but before `ValidatePublish` as we need to add the sourceId from the header before validating.